### ISSUE TITLE
Job scheduling performance regression

### ISSUE THREAD

**Issue by troykinsella** [concourse/concourse#1843]
# Bug Report

In a recent upgrade from `3.4.1` to `3.6.0`, performance has dropped significantly. Job execution times have increased by about 5 times. Jobs used to sit in pending state (all check marks checked) for up to 20 seconds, now up to 15 minutes. We have one ATC (m4.large) and 6 workers (i3.2xlarge). Each worker used to receive ~300 containers, but now only have ~160 containers each. General UI responsiveness has dropped as well. We had recreated and pruned all workers as part of the upgrade.

Seeing many new errors in the ATC:
```
 {"timestamp":"1511383796.178436279","source":"atc","message":"atc.container-collector.run.orphaned-containers.destroy-container.failed-to-destroy-database-container","log_level":2,"data":{"container":"d9ceae97-d47e-49e4-7d24-f42a69df1e4a","error":"pq: new row for relation \"volumes\" violates check constraint \"cannot_invalidate_during_initialization\"","session":"28.22.1.33","worker":"<redacted>"}}
```
... which I see is reported in #1823, but I'm not sure if it's related to these performance issues.

* Concourse version: 3.6.0
* Deployment type (BOSH/Docker/binary): binaries
* Infrastructure/IaaS: AWS
* Browser (if applicable): Chromium on Ubuntu
* Did this used to work? Yes.
* PostgreSQL: 9.5.2

**Comment by vito** [comment]
15 minutes sounds suspiciously close to our default retry logic cap when talking to a flaky worker.

Is this still happening?

The log quoted is probably a red herring. Do you see anything else that would explain the 15 minute delay? Maybe greb the logs for scheduling of the job?

Does adding another ATC help? They should roughly distribute the load.

**Comment by xtremerui** [comment]
@troykinsella have you tried upgrade the newer version of concourse? Do you still have this issue?

**Comment by troykinsella** [comment]
I have since upgraded to 3.8.0 and performance has been much better than what I reported above. Perhaps some important context is that I had dragged that concourse installation through many upgrades from around 1.0. By wiping the database and starting with a totally fresh installation, the reported performance issues are gone.

@vito That ~15 minute delay was very approximate. It could range from 10 to 20, but 15 being average, I suppose. I have yet to try a dual ATC setup, in which I could compare performance. But, I'm thinking my org will never be running pre-3.8.0 again, since 3.8.0+ has been relatively problem free for us.
